pkg/repository/v1/trigger.go

```diff
@@ -103,11 +103,19 @@ func newTriggerRepository(s *sharedRepository) TriggerRepository {
 	}
 }
 
-func (r *TriggerRepositoryImpl) TriggerFromEvents(ctx context.Context, tenantId string, opts []EventTriggerOpts) ([]*sqlcv1.V1Task, []*DAGWithData, error) {
+type EventDataForOlap struct {
+	EventKey                string
+	EventId                 string
+	TaskExternalId          string
+	EventAdditionalMetadata []byte
+	EventPayload            []byte
+}
+
+func (r *TriggerRepositoryImpl) TriggerFromEvents(ctx context.Context, tenantId string, opts []EventTriggerOpts) ([]*sqlcv1.V1Task, []*DAGWithData, []*EventDataForOlap, error) {
```

Comment:
Once we get to this many return types I think it's worth just returning a struct containing the arrays -- especially if we only ever return `nil, nil, nil, err` or all 3 arrays.

Reply:
I'm also curious if we just want to return a `map[*EventData][]string` which maps the event to a set of task external ids.